matrix-room.c

```diff
@@ -785,6 +920,41 @@ static const gchar *_get_my_display_name(PurpleConversation *conv)
 }
 
 /**
+ * Send an image message in a room
+ */
+void matrix_room_send_image(PurpleConversation *conv, int imgstore_id,
+        const gchar *message)
+{
+    JsonObject *content;
+    struct SendImageData *sid;
+
+    if (!imgstore_id)
+        return;
+    /* This is the hook_data on the event, it gets free'd by the event
+     * code when the event is free'd
+     */
+    sid = g_new0(struct SendImageData, 1);
+
+    /* We can't send this event until we've uploaded the image because
+     * the event contents including the file ID that we get back from
+     * the upload process.
+     * Our hook gets called back when we're ready to send the event,
+     * then we do the upload.
+     */
+    content = json_object_new();
+    json_object_set_string_member(content, "msgtype", "m.image");
+
+    sid->imgstore_id = imgstore_id;
+    sid->conv = conv;
+    purple_debug_info("matrixprpl", "%s: image id=%d\n", __func__, imgstore_id);
+    _enqueue_event(conv, "m.room.message", content, _send_image_hook, sid);
+    json_object_unref(content);
+    purple_conversation_write(conv, _get_my_display_name(conv),
```

Comment:
I wonder why I used `purple_conv_chat_write` in `matrix_room_send_message`. The commit comment is unenlightening, and I can't remember. Ah well. If you figure it out, let me know ;)